cmd/syncer/cmd/root.go

```diff
@@ -0,0 +1,131 @@
+	"k8s.io/component-base/logs"
+	"k8s.io/klog/v2"
+)
+
+const numThreads = 2
+
+func NewSyncerCommand() *cobra.Command {
+
+	options := NewDefaultOptions()
+	syncerCommand := &cobra.Command{
+		Use:   "syncer",
+		Short: "Synchronizes resources in `kcp` assigned to the clusters",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := options.validate(args)
+			return err
+		},
+	}
+	options.BindFlags(syncerCommand.Flags())
+	return syncerCommand
+}
+
+func (options *Options) validate(args []string) error {
+	//non-flag arguements
+	syncedResourceTypes := args
+	if len(syncedResourceTypes) == 0 {
+		syncedResourceTypes = []string{"deployments.apps"}
+		klog.Infoln("No resource types provided; using defaults.")
+	}
+	klog.Infof("Syncing the following resource types: %s", syncedResourceTypes)
+
+	// Create a client to dynamically watch "from".
+	if options.fromClusterName == "" {
+		return errors.New("--from_cluster is required")
+	}
+
+	fromConfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: options.fromKubeconfig}, nil).ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	var toConfig *rest.Config
+	if options.toKubeconfig != "" {
+		var toOverrides clientcmd.ConfigOverrides
+		if options.toContext != "" {
+			toOverrides.CurrentContext = options.toContext
+		}
+
+		toConfig, err = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			&clientcmd.ClientConfigLoadingRules{ExplicitPath: options.toKubeconfig},
+			&toOverrides).ClientConfig()
+		if err != nil {
+			return err
+		}
+	} else {
+		toConfig, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		return err
+	}
+
+	ctx := genericapiserver.SetupSignalContext()
+
+	if err := syncer.StartSyncer(ctx, fromConfig, toConfig, sets.NewString(syncedResourceTypes...), options.fromClusterName, options.pclusterID, numThreads); err != nil {
+		return err
+	}
+
+	<-ctx.Done()
+
+	return nil
+}
+
+type Options struct {
+	fromKubeconfig  string
+	fromClusterName string
+	toKubeconfig    string
+	toContext       string
+	pclusterID      string
+	Logs            *logs.Options
+}
+
+func NewDefaultOptions() *Options {
+	return &Options{
+		fromKubeconfig:  "",
+		fromClusterName: "",
+		toKubeconfig:    "",
+		toContext:       "",
+		pclusterID:      "",
+		Logs:            logs.NewOptions(),
+	}
+}
+
+func (options *Options) BindFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&options.fromKubeconfig, "from_kubeconfig", options.fromKubeconfig, "Kubeconfig file for -from cluster.")
+	fs.StringVar(&options.fromClusterName, "from_cluster", options.fromClusterName, "Name of the -from logical cluster.")
+	fs.StringVar(&options.toKubeconfig, "to_kubeconfig", options.toKubeconfig, "Kubeconfig file for -to cluster. If not set, the InCluster configuration will be used.")
+	fs.StringVar(&options.toContext, "to_context", options.toContext, "Context to use in the Kubeconfig file for -to cluster, instead of the current context.")
```

Comment:
Please convert the flag names to use dashes (`-`) to separate words instead of underscores (`_`)